### PR TITLE
[web] clean up dynamic calls, remove always_specify_types

### PR DESCRIPTION
**Pull request by yjbanov**
Remove the `avoid_dynamic_calls` exception in `web_ui`, and clean up all dynamic calls.

### CODE REVIEW

**Review by mdebbar — APPROVED** [review]

File: lib/web_ui/lib/src/engine/platform_dispatcher.dart [hunk 1]
```diff
@@ -535,7 +535,7 @@ class EnginePlatformDispatcher extends ui.PlatformDispatcher {
             if (renderer is CanvasKitRenderer) {
               assert(
                 decoded.arguments is int,
-                'Argument to Skia.setResourceCacheMaxBytes must be an int, but was ${decoded.arguments.runtimeType}',
+                'Argument to Skia.setResourceCacheMaxBytes must be an int, but was ${(decoded.arguments as Object?).runtimeType}',
```
Comment (ditman) [inline lib/web_ui/lib/src/engine/platform_dispatcher.dart:538]: I wonder why casting `decoded.arguments as Object?` doesn't make you access `runtimeType` trough `?.`... I guess null also has `runtimeType` so it doesn't matter?

**Review by ditman — APPROVED** [review]
Thanks for the cleanup!

File: lib/web_ui/lib/src/engine/platform_dispatcher.dart (hunk 1 above)
Comment (yjbanov) [reply lib/web_ui/lib/src/engine/platform_dispatcher.dart:538]: Yeah, `null` has all the `Object` methods, like `runtimeType`, `toString`, `operator==`, `hashCode`, etc. It's also surprising why `dynamic.runtimeType` is considered to be a dynamic invocation. Everything has a `runtimeType` in Dart. But I guess maybe `dynamic.anything` is considered "bad" by the lint.

Comment (matanlurey) [reply lib/web_ui/lib/src/engine/platform_dispatcher.dart:538]: This is a bug in the lint

Comment (matanlurey) [reply lib/web_ui/lib/src/engine/platform_dispatcher.dart:538]: Filed https://github.com/dart-lang/sdk/issues/55957.

**Pull request merged**